include/kernel.h

```diff
@@ -3419,7 +3419,7 @@ struct k_mem_block_id {
 	u32_t pool : 8;
 	u32_t level : 4;
 	u32_t block : 20;
-};
+} __aligned(sizeof(void *));
```

Comment:
(Huh... github just dropped another comment?  I swear I wrote and had this ready, but yet it wasn't posted.)

This struct is just an ID, passed around by value (mostly, though the API asks that the caller allocate it for historical reasons and takes a pointer) as a 32 bit word.  There's no need to align it, it stores no pointers and does not live in the heap itself.